src/webnn_native/openvino/GraphIE.cpp

```diff
@@ -1053,20 +1046,35 @@ namespace webnn_native { namespace ie {
         } else {
             attrs.shape_calculation_mode = ngraph_shape_calc_mode::Scales;
         }
-        const ngraph_node_t* axes =
-            AddConstantWithGraph<int64_t>(precision_e::I64, {4}, {0, 1, 2, 3});
-        ngraph_node_t* resampleNode;
-        if (transpose) {
-            input = TransposeInputLayout(input, true);
+        // axes: Interpolate layer only supports resize on spatial dimensions(depth, height and
+        // width)
+        // https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/mkldnn_interpolate_node.cpp#L2515
+        const ngraph_node_t* axesNode =
+            AddConstantWithGraph<int32_t>(precision_e::I32, {2}, {2, 3});
+        InputLayout inputLayout = InputLayout::Nchw;
```

Comment:
You can remove the definition of `InputLayout` and add a local variable named like "TransposeType transposeBackType = None;"
enum TransposeType { None, NhwcToNchw, HwncToNchw, NchwToNhwc, NchwToHwnc };
```
if (axes[0] == 0 && axes[1] == 1) {
      input = TransposeInputLayout(input, TransposeType::HwncToNchw);
     transposeBackType = NchwToHwnc;
  } else if (axes[0] == 1 && axes[1] == 2) {
      input = TransposeInputLayout(input, TransposeType::NhwcToNchw);
      transposeBackType = NchwToNhwc;
  }
```
You can only call like below when transposing back
```
if (transposeBackType != None) {
    resampleNode = TransposeInputLayout(resampleNode, transposeBackType);
}
```

Reply:
@fujunwei That's a good suggestion! I have optimized the code. PTAL, thanks a lot!